File: src/main/java/de/tectoast/emolga/commands/various/HelpCommand.java
package de.tectoast.emolga.commands.various;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import net.dv8tion.jda.api.EmbedBuilder;

import java.awt.*;
import java.util.stream.Collectors;

public class HelpCommand extends Command {

    public HelpCommand() {
        super("help", "Zeigt Hilfe über einen Command", CommandCategory.Various);
        this.otherPrefix = true;
        setArgumentTemplate(ArgumentManagerTemplate.builder().add("cmd", "Command", "Der Command, über den du Hilfe haben möchtest", ArgumentManagerTemplate.withPredicate("Command-Name", str -> byName(str) != null, false)).build());
    }

    @Override
    public void process(GuildCommandEvent e) {
        String cmdname = e.getArguments().getText("cmd");
        Command c = Command.byName(cmdname);

        EmbedBuilder builder = new EmbedBuilder();
        builder.setTitle(c.getPrefix() + c.getName());
        builder.setColor(Color.CYAN);
        ArgumentManagerTemplate template = c.getArgumentTemplate();
        builder.setDescription(c.getHelpWithoutCmd(e.getGuild()));
        builder.addField("Syntax", "```" + (template.hasSyntax() ? template.getSyntax() : c.getPrefix() + c.getName() + " "
                                                                                          + template.getArguments().stream().map(a -> (a.isOptional() ? "[" : "<") + a.getName() + (a.isOptional() ? "]" : ">")).collect(Collectors.joining(" "))) + "```", false);
        for (ArgumentManagerTemplate.Argument a : template.getArguments()) {
            builder.addField(a.getName(), (a.isOptional() ? "(Optional)\n" : "") + a.getHelp(), true);
        }
        builder.setFooter("Aufgerufen von " + e.getAuthor().getAsTag());
        if (template.hasExample()) {
            builder.addField("Beispiel", "```" + template.getExample() + "```", false);
        }
        if (!c.getAliases().isEmpty()) {
            builder.addField("Aliases", "`" + String.join("`, `", c.getAliases()) + "`", false);
        }
        e.reply(builder.build());
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/various/HelpCommand.kt
package de.tectoast.emolga.commands.various

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import net.dv8tion.jda.api.EmbedBuilder
import java.awt.Color

class HelpCommand : Command("help", "Zeigt Hilfe über einen Command", CommandCategory.Various) {
    init {
        otherPrefix = true
        argumentTemplate = ArgumentManagerTemplate.builder().add(
            "cmd",
            "Command",
            "Der Command, über den du Hilfe haben möchtest",
            ArgumentManagerTemplate.withPredicate("Command-Name", { str: String -> byName(str) != null }, false)
        ).build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val cmdname = e.arguments.getText("cmd")
        val c = byName(cmdname)!!
        val builder = EmbedBuilder()
        builder.setTitle(c.prefix + c.name)
        builder.setColor(Color.CYAN)
        val template = c.argumentTemplate
        builder.setDescription(c.getHelpWithoutCmd(e.guild))
        builder.addField(
            "Syntax",
            "```" + (if (template.hasSyntax()) template.syntax else c.prefix + c.name + " "
                    + template.arguments
                .joinToString(" ") { a: ArgumentManagerTemplate.Argument -> (if (a.isOptional) "[" else "<") + a.name + if (a.isOptional) "]" else ">" }) + "```",
            false
        )
        for (a in template.arguments) {
            builder.addField(a.name, (if (a.isOptional) "(Optional)\n" else "") + a.buildHelp(), true)
        }
        builder.setFooter("Aufgerufen von " + e.author.asTag)
        if (template.hasExample()) {
            builder.addField("Beispiel", "```" + template.example + "```", false)
        }
        if (c.aliases.isNotEmpty()) {
            builder.addField("Aliases", "`" + java.lang.String.join("`, `", c.aliases) + "`", false)
        }
        e.reply(builder.build())
    }
}